src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+
+/// <summary>
+/// Represents the request to the identity provider for logging in or provisioning a token.
+/// </summary>
+public sealed class InteractiveRequestOptions
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="InteractiveRequestOptions"/>.
+    /// </summary>
+    public InteractiveRequestOptions()
+    {
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="InteractiveRequestOptions"/> for signing in with the given return url and scopes.
+    /// </summary>
+    /// <param name="returnUrl">The URL to return to after the interactive operation.</param>
+    /// <param name="scopes">The scopes to request interactively.</param>
+    /// <returns>An <see cref="InteractiveRequestOptions"/> configured for signing in.</returns>
+    public static InteractiveRequestOptions SignIn(string returnUrl, IEnumerable<string> scopes)
```

Comment:
Could we do this and not need the doubles for each overload?

```suggestion
    public static InteractiveRequestOptions SignIn(string returnUrl, IEnumerable<string> scopes = null)
```

Reply:
I don't recall us using default values on parameters for APIs other than cancellation token. I generally avoid default parameter values on public APIs, but I am fine changing it if we are generally ok with it.